src/k8s-extension/azext_k8s_extension/_params.py

```diff
@@ -87,6 +87,10 @@ def load_arguments(self, _):
                    arg_group="Marketplace",
                    options_list=['--plan-publisher'],
                    help='The plan publisher is referring to the Publisher ID of the extension that is being taken from Marketplace portal under Usage Information + Support')
+        c.argument('skip_prerequisites',
+                   options_list=['--skip-prerequisites'],
+                   help='Specify whether to skip provisioning of prerequisites during extension create, or deprovisioning during extension delete.'
```

Comment:
>Specify whether to skip provisioning of prerequisites during extension create, or deprovisioning during extension delete. [](http://example.com/codeflow?start=25&length=121)

I am not sure how meaningful this info is for end user.  If this is only for ArcAutonomous, instead of this flag, can we detect the environment in the CLI and skip the prereqs internally?

Reply:
If we rely on a flag to be passed in by the user, if they don't, in the ArcAutonomous env., the pre-reqs will fail, and the experience wouldn't be good.
And, we can avoid this, if we detect and automatically do it without any input from the customer.

Reply:
Agree, we've considered the possibility of detecting the cloud type while there isn't a stable way to do that so far, such as checking against cloud name or profile. The current solution was discussed and chosen by Arc Autonomous finally.